File: LeetCode/706. Design HashMap/hashmap.ts
class MyHashMap<T> {
    dictionary: Record <number, T>
    length: number

    constructor() {
        this.dictionary = {}
        this.length = 0
    }

    put(key: number, value: T): void {
        this.dictionary[key] = value
    }

    get(key: number): T | -1 {
        return (this.dictionary[key] !== undefined) ? this.dictionary[key] : -1
    }

    remove(key: number): void {
        if (this.dictionary[key] !== undefined)
            delete this.dictionary[key]
    }
}

File: LeetCode/706. Design HashMap/hashmap.cs
public class MyHashMap {
    private Dictionary <int, int> hash;

    public MyHashMap() {
        hash = new Dictionary <int, int>();
    }
    
    public void Put(int key, int value) {
        hash[key] = value;
    }
    
    public int Get(int key) {
        if (hash.ContainsKey(key)) return hash[key];
        else return -1;
    }
    
    public void Remove(int key) {
        hash.Remove(key);
    }
}
